src/tests/roc_sndio/target_sox/test_player_recorder.cpp

```diff
@@ -115,41 +115,47 @@ TEST_GROUP(player_recorder) {
 };
 
 TEST(player_recorder, player_noop) {
-    MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 }
 
 TEST(player_recorder, player_error) {
-    MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 
     CHECK(!player.open("/bad/file"));
 }
 
 TEST(player_recorder, player_start_stop) {
     MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 
     core::TempFile file("test.wav");
     CHECK(player.open(file.path()));
 
-    player.start();
+    player.start(receiver);
     player.stop();
     player.join();
 }
 
 TEST(player_recorder, player_stop_start) {
     MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 
     core::TempFile file("test.wav");
     CHECK(player.open(file.path()));
 
     player.stop();
-    player.start();
+    player.start(receiver);
     player.join();
 }
 
+TEST(player_recorder, player_get_sample_rate) {
```

Comment:
I think we should test three cases:

* The user provided zero sample rate, SoX selected non-zero sample rate, Player was successfully opened.

* The user provided non-zero sample rate, SoX selected the same sample rate, Player was successfully opened.

* The user provided non-zero sample rate, but SoX selected some other sample rate. I think Player::open should fail in this case, because it's generally not a good idea to ignore what the user have asked explicitly.

I'm not sure if it's possible to add a test for the latter case because our tests operate with files instead of devices. However, we need to fix the behaviour anyway.

Reply:
Fixed